shm/test.cpp

```diff
@@ -0,0 +1,45 @@
+#include "cargo.hpp"
+#include "coordinates.hpp"
+#include "island.hpp"
+#include "map.hpp"
+#include "ship.hpp"
+#include "player.hpp"
+
+#include <vector>
+#include "gtest/gtest.h"
+
+
+class SHM_TestFixture: public ::testing::Test {
+public:
+    SHM_TestFixture( ) {
+        // initialization code here
+    }
+
+    void SetUp( ) {
+        // code here will execute just before the test ensues
+    }
+
+    void TearDown( ) {
+        // code here will be called just after the test completes
+        // ok to through exceptions from here if need be
+    }
+
+    ~SHM_TestFixture( )  {
+        // cleanup any pending stuff, but no exceptions allowed
+    }
+
+    // put in any custom data members that you need
+    Map testMap;
+};
+
+
+TEST_F(SHM_TestFixture, ShouldAlwaysPass) {
+
+    EXPECT_TRUE(true);
+}
+
+TEST_F(SHM_TestFixture, ShouldReturnNullPtrWhenAskedForIslandOutOfRange) {
+
+    Coordinates location(MAP_RND_DISTRIBUTION_MAX + 10, MAP_RND_DISTRIBUTION_MAX + 10);
+    EXPECT_EQ(nullptr, testMap.getIsland(location));
```

Comment:
I'm appreciated you try to write test :)

Check my Fixture, which can help you create own :)

```C++
class GameTests : public testing::Test {
public:
    void SetUp() override {
        time_ = std::make_unique<Time>();
        player_ = std::make_unique<Player>(kPlayerMoney, time_.get());
    }

    std::vector<Island> PrepareMapForTest() {
        std::vector<Island> vec;
        vec.reserve(3);
        vec.push_back(Island(kX1, kY1, time_.get()));
        vec.push_back(Island(kX2, kY2, time_.get()));
        vec.push_back(Island(kX3, kY3, time_.get()));
        return vec;
    }

    std::vector<std::unique_ptr<Cargo>> prepareCargoForTest() {
        std::vector<std::unique_ptr<Cargo>> cargo;
        cargo.push_back(std::make_unique<Alcohol>(kAlcoholAmount,
                                                  kAlcoholName, kAlcoholBasePrice, kAlcoholPercentage));
        cargo.push_back(std::make_unique<FruitMock>(kFruitAmount,
                                                    kFruitName, kFruitBasePrice, kFruitExpiryTime,
                                                    kFruitTimeElapsed, time_.get()));
        cargo.push_back(std::make_unique<Item>(kItemAmount,
                                               kItemlName, kItemBasePrice, Item::Rarity::legendary));
        return cargo;
    }

    std::unique_ptr<Time> time_;
    std::unique_ptr<Player> player_;
};
```

And also one test case

```C++
TEST_F(GameTests, ShouldCallNextDayWhenTimeIncrement) {
    auto fruit = std::make_unique<FruitMock>(kFruitAmount,
                                             kFruitName, kFruitBasePrice, kFruitExpiryTime,
                                             kFruitTimeElapsed, time_.get());
    auto* fruit_ptr = fruit.get();
    EXPECT_CALL(*fruit_ptr, GetAmount());
    player_->PurchaseCargo(std::move(fruit), 200);
    ASSERT_EQ(player_->GetCargo(0), fruit_ptr);

    EXPECT_CALL(*fruit_ptr, NextDay());
    ++*time_;
}
```

I hope it helps you :)